scripts/transcript_summarizer_aws_lambda.py

```diff
@@ -156,179 +248,197 @@ def extract_metadata_as_json(essay, chat_model=chat):
+            elif key == "Answer":
+                the_file.write("Answer: \n")
+                the_file.write(title_mark + "\n")
+                the_file.write(f"{value}\n\n")
+
+
+def lambda_handler(event, context):
+    # TODO implement
+
+    # Retrieve the source and destination bucket names from the event
+    source_bucket = event["Records"][0]["s3"]["bucket"]["name"]
+
+    # Retrieve the key (filename) of the object that triggered the Lambda
+    source_key = event["Records"][0]["s3"]["object"]["key"]
+
+    # ACCESS_KEY	= os.environ['ACCESS_KEY']
+    # SECRET_KEY	= os.environ['SECRET_KEY']
+    # s3_client = boto3.client('s3')
     s3_client = boto3.client(
-      's3',
-      aws_access_key_id=cfg.AWS_ACCESS_KEY,
-      aws_secret_access_key=cfg.AWS_SECRET_KEY
+        "s3", aws_access_key_id=ACCESS_KEY, aws_secret_access_key=SECRET_KEY
     )
-    
+
     print(source_bucket)
     print(source_key)
-    
+
+    # response = s3_client.get_object(Bucket=source_bucket, Key=source_key)
+    # docx_file = response['Body'].read()
+
+    # # Load the DOCX file using the python-docx library
+    # document = Document(io.BytesIO(docx_file))
+
+    # # Extract text from the document
+    # text = '\n'.join([paragraph.text for paragraph in document.paragraphs])
+
+    # return {
+    #     'statusCode': 200,
+    #     'body': json.dumps(text)
+    # }
+
     # Read the file content from the source bucket
     response = s3_client.get_object(Bucket=source_bucket, Key=source_key)
-    #file_content = response['Body'].read().decode('utf-8')
+    print("Response :", response)
+    # file_content = response['Body'].read().decode('utf-8')
     try:
-        raw_transcript = response['Body'].read().decode('utf-8')
+        print("In Try")
+        raw_transcript = response["Body"].read().decode("utf-8")
     except UnicodeDecodeError as e:
-        raw_transcript = response['Body'].read().decode('utf-8', errors='replace')
-        
-        # takes about 2-3 minutes to run
-    summary= full_transcript2essay(raw_transcript)
+        print("In exception", e)
+        raw_transcript = response["Body"].read().decode("utf-8", errors="replace")
+
+    # qa_index = raw_transcript.find("## QUESTION ANSWERING SECTION")
+
+    # Find the indices where each section starts
+    talk_index = raw_transcript.find("# TALK") + len("# TALK\n")
+    qa_index = raw_transcript.find("## QUESTION ANSWERING SECTION") + len(
+        "## QUESTION ANSWERING SECTION\n"
+    )
+
+    # Extract each section using slicing
+    first_part = raw_transcript[talk_index:qa_index].strip()
+    second_part = raw_transcript[qa_index:].strip()
+
+    # print("Raw Transcript : ",raw_transcript)
+    print("----first_part : ", first_part)
+    print("---- Second part : ", second_part)
+
+    # takes about 2-3 minutes to run
+    # summary= full_transcript2essay(raw_transcript)
+    summary = full_transcript2essay(first_part)  # considering only talk part
```

Comment:
now I can see the comment but I don't understand why you require this change and how it can be added.

the Essay is derived from this summary variable